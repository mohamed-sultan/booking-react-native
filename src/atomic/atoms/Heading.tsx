import React from 'react';
import {Text, StyleSheet} from 'react-native';
import { useThemeColors } from '@store/useTheme';

const Heading = ({children, style}: any) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return <Text style={[styles.heading, style]}>{children}</Text>;
};
export default React.memo(Heading);

const getStyles = (colors: any) => StyleSheet.create({
  heading: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '800',
    marginBottom: 25,
    color: colors.PrimaryText,
  },
});


