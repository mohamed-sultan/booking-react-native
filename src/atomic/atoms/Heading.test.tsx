import React from 'react';
import renderer from 'react-test-renderer';
import Heading from './Heading';

// Mock useThemeColors
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryText: '#000000',
  }),
}));

describe('Heading', () => {
  it('renders correctly with default style', () => {
    const tree = renderer
      .create(<Heading>Default Heading</Heading>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom style', () => {
    const tree = renderer
      .create(
        <Heading style={{ fontSize: 20, color: 'red' }}>
          Custom Styled Heading
        </Heading>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
