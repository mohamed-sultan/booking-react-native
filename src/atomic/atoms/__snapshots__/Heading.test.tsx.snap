// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly with custom style 1`] = `null`;

exports[`Heading renders correctly with default style 1`] = `null`;
